unlock-app/src/components/creator/lock/LockIconBar.js

```diff
@@ -105,30 +93,43 @@ LockIconBar.defaultProps = {
   edit: () => {},
 }
 
-const mapStateToProps = ({ transactions }, { lock }) => {
-  let withdrawalTransaction = null
-
-  Object.values(transactions).forEach(transaction => {
-    if (
-      transaction.type === TransactionType.WITHDRAWAL &&
-      transaction.lock === lock.address &&
-      transaction.confirmations < config.requiredConfirmations
-    )
-      withdrawalTransaction = transaction
+export const mapStateToProps = ({ transactions }, { lock }) => {
+  // Get all pending transactions as they will impact how we display the lock
+  const lockTransactions = Object.values(transactions)
+    .filter(transaction => {
+      return (
+        [transaction.lock, transaction.to].indexOf(lock.address) > -1 &&
+        (!transaction.confirmations ||
+          transaction.confirmations < config.requiredConfirmations)
+      )
+    })
+    .sort((t, u) => {
+      // We sort in reverse block order so we can get the "latest" transaction first
+      if (!t.blockNumber) {
```

Comment:
this would be great to extract into a utility, as I would use this on the paywall for sure!

Reply:
That is a neat idea for another PR!